src/test/libcephfs/test.cc

```diff
@@ -984,12 +984,23 @@ TEST(LibCephFS, PreadvPwritev) {
   struct iovec iovin[2];
   struct iovec iovout[2];
   ssize_t nwritten, nread;
+  char *out0, *out1, *in0, *in1; 
 
+  out0 = new char[str0.size()];
+  out1 = new char[str1.size()];
+  in0 = new char[str0.size()];
+  in1 = new char[str1.size()];
+
+  iovout[0].iov_base = out0;
```

Comment:
> I use a string to do the test, so c_str() is selected here. Hence, strcpy() is necessary.

i don't have a strong opinion regarding to `c_str()` versus `data()`. semantically, `data()` is more appropriate here, as you are using `str.size()`, where the ending `0` is not taken into consideration, and they are used as sample data instead of string for input and output.

> Hence, strcpy() is necessary.
you are **reading** from the strings. so no need to have a copy of them. so
```c
iovin[0].iov_base = str0.data();
```
would be simpler.

anyway, they are but cosmetic suggestions. it's up to you. =)